### PR TITLE
Add gs-adapter

### PR DESCRIPTION
**Pull request by kmcrawford**
This is the first commit of the google storage adapter.  This runs when I npm link.  
```
storage: {
    adapter: gs,
    options: {
      projectId: 'myproject-12345,
      bucket: 'my_bucket',
      path: '//storage.googleapis.com/my_bucket/',
      componentsDir: 'components',
    }
  },
```

I have a todo, where I don't quite understand the callback logic, but this is working locally.

### CODE REVIEW

File: packages/oc-gs-storage-adapter/__test__/gs.test.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
+'use strict';
+const gs = require('../');
+const Storage = require('@google-cloud/storage');
+
+// //Mock AWS functions
```
Comment (nickbalestra) [inline packages/oc-gs-storage-adapter/__test__/gs.test.js]: We reworked and improved the tests of the oc-s3-storage-adapter, perhapsh is worth to sync [those](https://github.com/opencomponents/storage-adapters/blob/master/packages/oc-s3-storage-adapter/__test__/s3.test.js) ones?

File: packages/oc-gs-storage-adapter/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "url": "https://github.com/opencomponents/storage-adapters.git"
+  },
+  "bugs": {
+    "url": "https://github.com/opencomponents/storage-adapters/issues"
+  },
+  "homepage": "https://github.com/opencomponents/storage-adapters#readme",
+  "keywords": [
+    "oc",
+    "opencomponents",
+    "s3"
```
Comment (nickbalestra) [inline packages/oc-gs-storage-adapter/package.json]: I think this should be gs?

File: packages/oc-s3-storage-adapter/package.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
   "keywords": [
     "oc",
     "opencomponents",
-    "s3"
+    "goolge storage",
+    "gs"
```
Comment (nickbalestra) [inline packages/oc-s3-storage-adapter/package.json]: ah see previous comment, I guess for some reason those are the changes you wanted to be in the gs, not on the s3 package, probably something went wrong when checking in the files

File: packages/oc-gs-storage-adapter/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "oc-gs-storage-adapter",
+  "version": "1.0.0",
```
Comment (nickbalestra) [inline packages/oc-gs-storage-adapter/package.json]: Maybe we can set the version to 0.0.1 to start with? as discussed I'll try to publish this as a [prerelease](https://www.npmjs.com/package/lerna#--npm-tag-tagname) (`beta`)

**Comment by kmcrawford** [comment]
I have re-factored the code to work with the test cases.  Some of things aren't needed in gs  (contentType, & etc) as they are in aws, but I added the data for consistency. (except for file content, if needed I can add)

**Comment by nickbalestra** [comment]
I'll add my self the link to the main readme. Thanks for that @kmcrawford

**Pull request merged**